src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,211 @@
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        return carsModels;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+        updateDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car from DB by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id"
+                + " FROM cars c"
+                + " JOIN cars_drivers cd"
+                + " ON c.id = cd.driver_id"
+                + " WHERE cd.car_id = ? AND c.is_deleted = FALSE;";
+        List<Car> carsModels = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturer(resultSet);
+                List<Driver> drivers = getDriversForCar(car.getId());
+                car.setDrivers(drivers);
+                carsModels.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars from DB by driver id: "
+                    + driverId, e);
+        }
+        return carsModels;
+    }
+
+    private Car parseCarWithManufacturer(ResultSet resultSet) throws SQLException {
+        Car car = new Car(resultSet.getObject("car_id", Long.class),
+                resultSet.getString("model"),
+                new Manufacturer(
+                        resultSet.getObject("manufacturer_id", Long.class),
+                        resultSet.getString("name"),
+                        resultSet.getString("country")));
+        return car;
```

Comment:
interestingly shortened! But you can shorten more (optionally)
```suggestion
        return new Car(resultSet.getObject("car_id", Long.class),
                resultSet.getString("model"),
                new Manufacturer(
                        resultSet.getObject("manufacturer_id", Long.class),
                        resultSet.getString("name"),
                        resultSet.getString("country")));
```

Reply:
ok